loopchain/channel/channel_inner_service.py

```diff
@@ -490,37 +491,47 @@ def __add_tx_list(self, tx_list):
 
             new_block_height = subscriber_block_height + 1
             new_block = blockchain.find_block_by_height(new_block_height)
-            confirm_info: str = blockchain.find_confirm_info_by_height(new_block_height).decode("utf-8")
+            confirm_info: bytes = blockchain.find_confirm_info_by_height(new_block_height)
 
             if new_block is None:
                 logging.warning(f"Cannot find block height({new_block_height})")
                 await asyncio.sleep(0.5)  # To prevent excessive occupancy of the CPU in an infinite loop
                 continue
 
             logging.debug(f"announce_new_block: height({new_block.header.height}), hash({new_block.header.hash}), "
-                          f"target: {self._citizen_set}")
+                          f"target: {self._citizens}")
             bs = BlockSerializer.new(new_block.header.version, blockchain.tx_versioner)
             return json.dumps(bs.serialize(new_block)), confirm_info
 
     @message_queue_task
-    async def register_subscriber(self, peer_id):
-        if len(self._citizen_set) >= conf.SUBSCRIBE_LIMIT:
+    async def register_citizen(self, peer_id, target, connected_time):
+        if len(self._citizens) >= conf.SUBSCRIBE_LIMIT:
             return False
         else:
-            self._citizen_set.add(peer_id)
-            logging.info(f"register new subscriber: {peer_id}")
-            logging.debug(f"remaining all subscribers: {self._citizen_set}")
+            Citizen = namedtuple("Citizen", "peer_id target connected_time")
```

Comment:
How about moving this Citizen declaration to top of this file to use it on type hint?
Ultimately, it should be placed in a separate location later.

Reply:
CitizenInfo(namedtuple ) has been defined as member variable to use it inside methods and type hint. Thanks for your help:)